### PR TITLE
SCALRCORE-19437 IaCP Provider > RBAC > Roles & Access Policies resources and datasources

### PR DESCRIPTION
**Pull request by vaniakov**
### Changelog
* [x] I have updated CHANGELOG.md according to [CONTRIBUTING.md](../CONTRIBUTING.md)

### Documentation
* [x] I have updated resource/data source documentation in [docs](../docs)

### State
* [ ] My changes affect the state
* [ ] I have included a state migration and a unit test for it

### CODE REVIEW

File: scalr/testing.go [hunk 1]
```diff
@@ -8,6 +8,9 @@ import (
 )
 
 const defaultAccount = "acc-svrcncgh453bi8g"
+const testUser = "user-suh84u6vuvidtbg"     // test@scalr.com
+const readOnlyRole = "role-t67mjtmabulckto" // Reader
+const userRole = "role-t67mjtmauajto7g"     // Reader
```
Comment (vaniakov) [inline scalr/testing.go:13]: bad comment

File: scalr/resource_scalr_access_policy.go [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+	}
+
+	switch scopeType {
+	case "workspace":
+		options.Workspace = &scalr.Workspace{ID: scopeId}
+	case "environment":
+		options.Environment = &scalr.Environment{ID: scopeId}
+	case "account":
+		options.Account = &scalr.Account{ID: scopeId}
+	}
```
Comment (petroprotsakh) [inline scalr/resource_scalr_access_policy.go:155]: @vaniakov для значень subject і scope вже є стрічкові константи, варто юзати їх замість хардкоду. `string(Workspace)`, або навіть дати `func (s Scope) String() string`

**Review by petroprotsakh — APPROVED** [review]
looks good

File: scalr/resource_scalr_access_policy.go [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+	User           Subject = "user"
+	Team           Subject = "team"
+	ServiceAccount Subject = "service_account"
+)
+
+const (
+	Workspace   Scope = "workspace"
+	Environment Scope = "environment"
+	Account     Scope = "account"
+)
```
[the remark this answers is not in the document]
Comment (vaniakov) [reply scalr/resource_scalr_access_policy.go:25]: поки для них не знайшов кращого місця

File: scalr/resource_scalr_access_policy.go (hunk 2 above)
Comment (vaniakov) [reply scalr/resource_scalr_access_policy.go:155]: @petroprotsakh makes sense

**Review by vmotso — APPROVED** [review]

File: docs/resources/scalr_access_policy.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+layout: "scalr"
+page_title: "Scalr: scalr_access_policy"
+sidebar_current: "docs-resource-scalr-access-policy"
+description: |-
+  Manages access policies.
+---
+
+# scalr_access_policy Resource
+
+Manage the Scalr IAM access policies. Create, update and destroy
```
Comment (rtfee) [inline docs/resources/scalr_access_policy.md:11]: Add a "." at the end: Create, update and destroy.

File: docs/resources/scalr_role.md [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+layout: "scalr"
+page_title: "Scalr: scalr_role"
+sidebar_current: "docs-resource-scalr-role"
+description: |-
+  Manages role.
+---
+
+# scalr_role Resource
+
+Manage the Scalr IAM roles. Create, update and destroy
```
Comment (rtfee) [inline docs/resources/scalr_role.md:11]: Add a "." at the end: Create, update and destroy.

File: docs/resources/scalr_role.md [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+* `account_id` - (Required) ID of the account.
+* `permissions` - (Required) Array of permission names.
+* `description` - (Optional) Verbose description of the role.
+
+## Attribute Reference
+
+All arguments plus:
+
+* `id` - The ID of the role.
+* `is_system` - Boolean indicates if the role can be edited.
```
Comment (rtfee) [inline docs/resources/scalr_role.md:43]: Maybe add at the end: "System roles are maintained by Scalr and cannot be changed"

**Pull request merged**